cmd/prometheus/main.go

```diff
@@ -667,6 +670,12 @@ func main() {
 
 			cfg.tsdb.MaxBlockDuration = maxBlockDuration
 		}
+
+		// Delayed compaction checks
+		if cfg.tsdb.EnableDelayedCompaction && (cfg.tsdb.CompactionDelayMaxPercent >= 100 || cfg.tsdb.CompactionDelayMaxPercent < 0) {
+			cfg.tsdb.CompactionDelayMaxPercent = tsdb.DefaultCompactionDelayMaxPercent
+			logger.Warn("The --storage.tsdb.delayed-compaction.max-percent should have a value between 0 and 100. Using default 10%")
```

Comment:
```suggestion
			logger.Warn("The --storage.tsdb.delayed-compaction.max-percent should have a value between 0 and 100. Using default", "default", tsdb.DefaultCompactionDelayMaxPercent)
```
(nit: if we could put it before the assignment it'd be even better)